src/NexusMods.Paths/AbsolutePath.cs

```diff
@@ -84,19 +85,34 @@ internal AbsolutePath(string? directory, string fileName, IFileSystem fileSystem
     /// </remarks>
     public static AbsolutePath FromFullPath(string fullPath, IFileSystem? fileSystem = null)
     {
-        var index = fullPath.LastIndexOf(Path.DirectorySeparatorChar);
-        if (index < 0)
-        {
-            // No directory
+        var span = fullPath.AsSpan();
+
+        // path is not rooted
+        var rootLength = GetRootLength(span);
+        if (rootLength == 0)
             return new AbsolutePath(null, fullPath, fileSystem ?? Paths.FileSystem.Shared);
+
+        // path is only the root directory
+        if (span.Length == rootLength)
+            return new AbsolutePath(fullPath, "", fileSystem ?? Paths.FileSystem.Shared);
```

Comment:
Can you hoist `fileSystem ?? Paths.FileSystem.Shared` to a variable; i.e. `fileSystem ??= Paths.FileSystem.Shared;`, and then use `fileSystem` below. This should save a bit of runtime by making the branches over the constructors faster due to less instructions total generated.